compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/ComposeScene.kt

```diff
@@ -0,0 +1,267 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.scene
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionContext
+import androidx.compose.runtime.CompositionLocal
+import androidx.compose.runtime.CompositionLocalContext
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.InternalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusDirection
+import androidx.compose.ui.focus.FocusManager
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusProperties
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.Canvas
+import androidx.compose.ui.input.key.KeyEvent
+import androidx.compose.ui.input.key.KeyInputModifierNode
+import androidx.compose.ui.input.pointer.InteropViewCatchPointerModifier
+import androidx.compose.ui.input.pointer.PointerButton
+import androidx.compose.ui.input.pointer.PointerButtons
+import androidx.compose.ui.input.pointer.PointerEventType
+import androidx.compose.ui.input.pointer.PointerKeyboardModifiers
+import androidx.compose.ui.input.pointer.PointerType
+import androidx.compose.ui.node.LayoutNode
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.platform.setContent
+import androidx.compose.ui.semantics.SemanticsNode
+import androidx.compose.ui.semantics.SemanticsOwner
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.IntRect
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.Popup
+import org.jetbrains.skiko.currentNanoTime
+
+/**
+ * Represents a static [CompositionLocal] key for a [ComposeScene] in Jetpack Compose.
+ *
+ * @see ComposeScene
+ */
+internal val LocalComposeScene = staticCompositionLocalOf<ComposeScene?> { null }
+
+/**
+ * The local [ComposeScene] is typically not-null. This extension can be used in these cases.
+ */
+@Composable
+internal fun CompositionLocal<ComposeScene?>.requireCurrent(): ComposeScene {
+    return current ?: error("CompositionLocal LocalComposeScene not provided")
+}
+
+/**
+ * A virtual container that encapsulates Compose UI content. UI content can be constructed via
+ * [setContent] method and with any Composable that manipulates [LayoutNode] tree.
+ *
+ * After [ComposeScene] will no longer needed, you should call [close] method, so all resources
+ * and subscriptions will be properly closed. Otherwise, there can be a memory leak.
+ */
```

Comment:
```suggestion
 * It is marked as @InternalComposeUiApi and used by default Compose entry points
 * (application, runComposeUiTest, ComposeWindow). It can also be used by
 * external users for embedding Compose to other platforms, but there is no 
 * stability guranties.
 */
```

We should describe why we it is public